finder-app/writer.c

```diff
@@ -0,0 +1,86 @@
+/*****************************************************************************
+​*​ ​Copyright​ ​(C)​ ​2023 ​by​ Matthew Skogen
+​*
+​*​ ​Redistribution,​ ​modification​ ​or​ ​use​ ​of​ ​this​ ​software​ ​in​ ​source​ ​or​ ​binary
+​*​ ​forms​ ​is​ ​permitted​ ​as​ ​long​ ​as​ ​the​ ​files​ ​maintain​ ​this​ ​copyright.​ ​Users​ ​are
+​*​ ​permitted​ ​to​ ​modify​ ​this​ ​and​ ​use​ ​it​ ​to​ ​learn​ ​about​ ​the​ ​field​ ​of​ ​embedded
+​*​ ​software.​ ​Matthew Skogen ​and​ ​the​ ​University​ ​of​ ​Colorado​ ​are​ ​not​ ​liable​ ​for
+​*​ ​any​ ​misuse​ ​of​ ​this​ ​material.
+​*
+*****************************************************************************/
+/**
+​*​ ​@file​ writer.c
+​*​ ​@brief​ Application entry point
+​*
+​*​ ​@author​s ​Matthew Skogen
+​*​ ​@date​ January 28 ​2023
+*
+*  Replacement for writer.sh, but:
+*   1. Doesn't need to create directories that don't exist
+*   2. Setup syslog logging using LOG_USER facility.
+*   3. Use syslog to write a message "Writing <string> to <file>" at LOG_DEBUG
+*       - string = text written to file
+*       - file = file created by the script
+*   4. Use syslog to log unexpected errors with LOG_ERR level
+​*/
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <syslog.h>
+
+int main(int argc, char *argv[])
+{
+    /* Open connection to system logger with default LOG_USER facility */
+    openlog("writer", LOG_CONS, LOG_USER);
+
+    /* Verify proper number of arguments passed in */
+    if (argc != 3) {
+        printf("ERROR: Invalid number of arguments %i\n", argc);
+        printf("Usage: ./writer /path/to/new/file <write_string>\n");
+        return 1;
+    }
+
+    /* First argument is a path to directory on filesystem */
+    char *write_file=argv[1];
+
+    /* Second argument is text string to be written to the write file */
+    char *write_str=argv[2];
```

Comment:
I like the idea to save to variables.  Anything we know about whether these should be modified?  Any way to have the compiler help us enforce?